### PR TITLE
PCA20035 support only newest hardware revision

### PR DESCRIPTION
**Pull request by jtguggedal**
This PR contains two commits that removes support for earlier hardware versions of nrf52840_pca20035 and nrf9160_pca20035. From now on, only the latest hardware version is supported. This simplifies the board file structure.

The third commit removes configurations in the asset_tracker application that were specific to the deprecated board versions.

### CODE REVIEW

**Comment by hmhalvorsen** [comment]
Overall this looks good, although it can't be tested in its current state. As we've discussed earlier could it be an idea to have the adp536x source code move from board_nonsecure.c to board_secure.c in a separate commit.

**Comment by jtguggedal** [comment]
> Overall this looks good, although it can't be tested in its current state. As we've discussed earlier could it be an idea to have the adp536x source code move from board_nonsecure.c to board_secure.c in a separate commit.

I've split the ADP536X into a separate commit. Also added a commit enabling ADP536X buck regulator's discharge resistor by default.

File: boards/arm/nrf9160_pca20035/CMakeLists.txt [hunk 1]
```diff
@@ -4,12 +4,11 @@
...
 
-zephyr_library()
+if(CONFIG_TRUSTED_EXECUTION_SECURE AND NOT DEFINED CONFIG_MCUBOOT)
+	zephyr_library()
+	zephyr_library_sources(board_secure.c)
+endif()
 
 zephyr_library_sources_ifdef(
 	CONFIG_TRUSTED_EXECUTION_NONSECURE
 	board_nonsecure.c)
```
Comment (SebastianBoe) [inline boards/arm/nrf9160_pca20035/CMakeLists.txt]: You need to make sure there is no code path where this file is added to a library that hasn't been created.

Comment (jtguggedal) [reply boards/arm/nrf9160_pca20035/CMakeLists.txt]: I'm not very familiar with cmake, should these source files be added some other way? Or is this the correct way to do it?

Comment (ioannisg) [reply boards/arm/nrf9160_pca20035/CMakeLists.txt]: Could we, instead, use the BOARD names, to compile-in these source files?

Comment (jtguggedal) [reply boards/arm/nrf9160_pca20035/CMakeLists.txt]: @ioannisg Yes, that sounds like a better idea. As discussed elsewhere, `CONFIG_TRUSTED_EXECUTION_SECURE` will no longer be set in defconfig to align with PCA10090, so good idea to use board name instead to distinguish the two.

Comment (SebastianBoe) [reply boards/arm/nrf9160_pca20035/CMakeLists.txt]: @jtguggedal : I suspect that the CMake code is written incorrectly, but I can't be sure without knowing what kind of behaviour is desired. If you could express in english what you want to happen then I can translate that to CMake.

Comment (jtguggedal) [reply boards/arm/nrf9160_pca20035/CMakeLists.txt]: @SebastianBoe I'll try: I want to compile in `board_secure.c` and `board_secure.c` depending on the Kconfig option. They're supposed to run during bootup, setting up the board control ICs, modem MAGPIO settings and such.

Comment (SebastianBoe) [reply boards/arm/nrf9160_pca20035/CMakeLists.txt]: Could they both be enabled simultaneously?

Could neither be enabled?

Comment (jtguggedal) [reply boards/arm/nrf9160_pca20035/CMakeLists.txt]: `board_secure.c` should be run once every time the board boots, as it controls the onboard PMIC. `board_nonsecure.c` should only run for non-secure applications compiled for `nrf9160_pca20035ns` target

Comment (SebastianBoe) [inline boards/arm/nrf9160_pca20035/CMakeLists.txt]: If I understand you correctly, then you should change

zephyr_library_sources_ifdef(
  CONFIG_TRUSTED_EXECUTION_NONSECURE

into

if(CONFIG_TRUSTED_EXECUTION_NONSECURE)

and put a second zephyr_library() invocation into that if-statement (before the "sources" invocation). 

This is to ensure that when CONFIG_TRUSTED_EXECUTION_SECURE, we don't add board_secure.c to some unrelated library that has been created earlier.

Comment (jtguggedal) [reply boards/arm/nrf9160_pca20035/CMakeLists.txt]: Thanks! I updated according to your suggestion, and it seems to be working well

File: boards/arm/nrf52840_pca20035/Kconfig.board [hunk 2]
```diff
@@ -6,12 +6,7 @@
 
 if SOC_NRF52840_QIAA
```
[the remark this answers is not in the document]
Comment (jtguggedal) [reply boards/arm/nrf52840_pca20035/Kconfig.board]: Fixed

File: boards/arm/nrf52840_pca20035/nrf52840_pca20035.dts [hunk 3]
```diff
@@ -5,6 +5,90 @@
...
+	current-speed = <1000000>;
+	status = "ok";
+	tx-pin = <25>;
+	rx-pin = <32>;
+	rts-pin = <22>;
+	cts-pin = <19>;
+};
+
+&usbd {
+	compatible = "nordic,nrf-usbd";
```
[the remark this answers is not in the document]
Comment (jtguggedal) [reply boards/arm/nrf52840_pca20035/nrf52840_pca20035.dts]: Fixed

File: boards/arm/nrf9160_pca20035/nrf9160_pca20035_defconfig [hunk 4]
```diff
@@ -27,5 +25,13 @@ CONFIG_UART_0_NRF_UARTE=y
 CONFIG_CONSOLE=y
 CONFIG_UART_CONSOLE=y
 
+# Enable I2C
+CONFIG_I2C=y
+CONFIG_I2C_NRFX=y
```
[the remark this answers is not in the document]
Comment (jtguggedal) [reply boards/arm/nrf9160_pca20035/nrf9160_pca20035_defconfig]: Fixed

File: boards/arm/nrf52840_pca20035/nrf52840_pca20035_defconfig [hunk 5]
```diff
@@ -2,17 +2,15 @@ CONFIG_ARM=y
...
-# PCA20035 v0.3.0 is set as default board version
-CONFIG_BOARD_NRF52840_PCA20035_V0_3_0=y
+CONFIG_BOARD_NRF52840_PCA20035=y
 
 # Enable MPU
 CONFIG_ARM_MPU=y
 
 # enable uart driver
 CONFIG_SERIAL=y
 CONFIG_UART_NRFX=y
```
[the remark this answers is not in the document]
Comment (jtguggedal) [reply boards/arm/nrf52840_pca20035/nrf52840_pca20035_defconfig]: Thanks for the review, fixed according to your comments

**Comment by jtguggedal** [comment]
It looks like the CI is green on the last MCUboot check, but it's not reported for some reason.
@rlubos / @ioannisg  Could you have a look on this PR and see if you're ok with the state of it?

**Comment by ioannisg** [comment]
> It looks like the CI is green on the last MCUboot check, but it's not reported for some reason.
> @rlubos / @ioannisg Could you have a look on this PR and see if you're ok with the state of it?

I'll take a look.
To me it seems CI is still running, no?

**Comment by jtguggedal** [comment]
> > It looks like the CI is green on the last MCUboot check, but it's not reported for some reason.
> > @rlubos / @ioannisg Could you have a look on this PR and see if you're ok with the state of it?
> 
> I'll take a look.
> To me it seems CI is still running, no?

@ioannisg Yes, that's what's reported on GitHub, but inside Jenkins it's all green. @thst-nordic Do you know what's going on with this one?

**Comment by rlubos** [comment]
@jtguggedal We need to integrate #714 first, right? I think both PRs have enough approvals, wanted to merge them yesterday, but the CI...

**Comment by jtguggedal** [comment]
> @jtguggedal We need to integrate #714 first, right? I think both PRs have enough approvals, wanted to merge them yesterday, but the CI...

@rlubos You're right. I somehow thought that went in yesterday, but sure didn't.. CI is happy with it now.

**Comment by rlubos** [comment]
@jtguggedal #714 is now merged, you can rebase and we'll integrate this one next.

**Comment by jtguggedal** [comment]
> @jtguggedal #714 is now merged, you can rebase and we'll integrate this one next.

@rlubos Thanks. Rebased now.

**Comment by jtguggedal** [comment]
Looks like CI passes now.

**Comment by rlubos** [comment]
So @ioannisg, do you plan to review or can we merge already?

**Pull request merged**